towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+                             project.scm_url == scm_url and project.scm_branch == branch_name]
+        if len(matching_projects) > 0:
+            for project in matching_projects:
+                self._logger.info("Updating the project: {}".format(project.name))
+                project_update = self.update_project_by_id(project.id)
+                project_update_id_list.append(project_update['id'])
+            self._logger.info("Updated {} project(s) with the url '{}' and branch name '{}'".
+                              format(len(matching_projects), scm_url, branch_name))
+            return project_update_id_list
+        else:
+            self._logger.warning("No project was updated. Reason: no tower project found with the given "
+                                 "git url '{}' and branch '{}'".format(scm_url, branch_name))
+            return []
+
+    def get_all_job_templates(self):
+        """Get all the job templates of the ansible tower.
+
+        Returns:
+            list: the list of all job templates as EntityManager object.
+
+        """
+        job_templates = [item for item in self.job_templates]
+        return job_templates
+
+    def get_job_templates_by_project(self, project_name):
+        """Get all the job templates for a given project name.
+
+        Args:
+            project_name: the complete api url of the project.
+
+        Returns:
+            list: a list of all the job templates object for the given project name.
+
+        """
+        required_job_templates = []
+        job_templates = self.get_all_job_templates()
+        for job_template in job_templates:
+            if job_template.project.name == project_name:
+                required_job_templates.append(job_template)
+        return required_job_templates
+
+    def change_job_template_data(self, job_template_data):
+        """Send API PATCH request to update the job template information with the given data.
+        https://docs.ansible.com/ansible-tower/3.6.1/html/towerapi/api_ref.html#/Authentication/Authentication_applications_partial_update_0
+
+        Args:
+            job_template_data: updated data of the job template as dictionary
+
+        Returns:
+            list: List of response of api request as json on success, False otherwise.
+
+        """
+        job_template_id = job_template_data['id']
+        job_url = '{api}/job_templates/{id}/'.format(api=self.api, id=job_template_id)
+        response = self.session.patch(job_url, data=json.dumps(job_template_data))
+        if not response.ok:
+            self._logger.error("Error updating the job template with the given data '{}'".format(job_template_data))
+            return None
+        else:
+            self._logger.info("Job template with the name '{}' is updated successfully.".format(job_template_data['name']))
+            return response.json()
+
+    def change_project_of_job_template(self, job_template, project_id):
```

Comment:
Please refactor this to be a setter of the project property of the JobTemplate object. Make it use a project name as an input argument and not an ID.

Reply:
Implemented with b0db88a5e65f1ce0a8c96b5f5a35af546d5d2658

Project of the job template has to be updated with the ID of the project as the value of project in the API reference is set to ID.

![image](https://user-images.githubusercontent.com/18288421/121660745-cfd11c00-caa3-11eb-8043-75b10994293f.png)